### PR TITLE
feat: Added delete and reauthenticate tests to user_repository

### PR DESCRIPTION
**Pull request by msusviela**

### CODE REVIEW

File: somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@ import 'dart:async';
 
 import 'package:api_client/api_client.dart';
 import 'package:auth_client/auth_client.dart';
+import 'package:flutter/foundation.dart';
```
Comment (gianfrancopapa) [inline somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart]: Is this necessary?

File: somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart [hunk 2]
```diff
@@ -18,7 +19,9 @@ class UserRepository {
   final ApiClient _apiClient;
   final AuthenticationClient _authenticationClient;
 
-  Future<User?> _onAuthStatusChanged(AuthenticationEvent event) async {
+  /// Maps the [AuthenticationEvent] to a [User] object.
+  @visibleForTesting
+  Future<User?> onAuthStatusChanged(AuthenticationEvent event) async {
```
Comment (gianfrancopapa) [inline somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart:22]: Can we keep this as private?

File: somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart (hunk 1 above)
Comment (msusviela) [reply somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart]: Yes, it's for the visibleForTesting notation

File: somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart (hunk 2 above)
Comment (msusviela) [reply somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart:22]: We need it so we can test it and achieve higher coverage

File: somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart (hunk 1 above)
Comment (gianfrancopapa) [reply somnio_starter/__brick__/{{project_name.snakeCase()}}/packages/user_repository/lib/src/user_repository.dart]: Ohhh ok, let's keep the method as public then so we don't add flutter as a dependency

**Review by gianfrancopapa — APPROVED** [review]

**Pull request merged**